nuxeo-services/nuxeo-platform-web-common/src/main/resources/web/nuxeo.war/login.jsp

```diff
@@ -526,35 +536,40 @@ a.mobileAppLink:hover {
     <fmt:message bundle="${messages}" key="label.mobile.openInApp" />
   </a>
 </div>
-<% } %>
 
 <script type="text/javascript">
-  <% if (displayMobileBanner) { %>
   // Build mobile app links if the mobile banner is displayed
-  var indexOfHash = window.location.href.indexOf('#!');
-  if (indexOfHash > -1) {
-    // lastPart = #!/doc/default/f6fa9686-3618-47a8-9419-ff1cc76fc857
-    // or lastPart = #!/doc/f6fa9686-3618-47a8-9419-ff1cc76fc857
-    var lastPart = window.location.href.substring(indexOfHash);
-
-    var docPart;
-    var parts = lastPart.split('/');
+  <% if (requestedUrlFragment != null) { %>
+
+  var urlFragment = decodeURIComponent('<%=requestedUrlFragment%>');
+  var docPart;
+  if (urlFragment.startsWith('!/browse')) {
+    // no repository name for 'browse' URL
+    docPart = urlFragment.replace('!/browse/', 'default/path/');
```

Comment:
Why not fetch the default repository name from the `RepositoryManager` as on master?